datahub-web-react/src/app/entity/shared/containers/profile/header/EntityTabs.tsx

```diff
@@ -25,23 +25,22 @@ const Tab = styled(Tabs.TabPane)`
 `;
 
 export const EntityTabs = <T,>({ tabs, selectedTab }: Props) => {
-    const { entityData } = useEntityData();
+    const { entityData, loading } = useEntityData();
     const routeToTab = useRouteToTab();
     const baseEntity = useBaseEntity<T>();
 
+    const visibleTabs = tabs.filter((tab) => tab.display?.visible(entityData, baseEntity));
+    const enabledAndVisibleTabs = visibleTabs.filter((tab) => tab.display?.enabled(entityData, baseEntity));
+
     useEffect(() => {
-        if (!selectedTab) {
-            if (tabs[0]) {
-                routeToTab({ tabName: tabs[0].name, method: 'replace' });
-            }
+        if (!loading && !selectedTab && enabledAndVisibleTabs[0]) {
```

Comment:
Starting to wonder if this 'loading' is even necessary at all. 

This useEffect will run whenever it's inputs change. In this case, selectedTab will change once a new Tab is selected (e.g. the visible / enabled tabs change. 

Therefore, you shouldn't really need to watch the loading parameter of the entity, right? It can be made a bit simpler?

Reply:
We have to delay the first routing (when selectedTab is still undefined) until the data is loaded, if we would not delay it, the "wrong" tab could be selected after the data was loaded (see also the edit of my previous wall of text 😇).